gazelle_cabal/dependency_resolution.go

```diff
@@ -408,7 +409,51 @@ func mapSortedStringKeys(m map[string]bool) []string {
 		i++
 	}
 	sort.Strings(ss)
-	return ss
+	s_result := removeEquivalent(samePackage, ss)
+	return s_result
+}
+
+// This function assumes that equivalent elements are bundled and only keeps the last element of each equivalence class.
+func removeEquivalent(equiv func(string, string) bool, s []string) []string {
+	if len(s) < 1 {
+		return s
+	}
+
+	first_free_position := 1
+	for curr := 1; curr < len(s); curr++ {
+		if equiv(s[first_free_position-1], s[curr]) {
+			// If the last added element is equivalent is considered one,
+			// then we move the cursor back so that the currently considered element
+			// replace the previously added one.
+			// This function is only keeping the last representative of the equivalence class
+			// because it is used to deal with packages having or not a version number,
+			// and in case both option are available, the version  number should be kept.
+			first_free_position--
+		}
+		s[first_free_position] = s[curr]
+		first_free_position++
+	}
+
+	return s[:first_free_position]
+}
+
+func samePackage(s1 string, s2 string) bool {
+	ss1 := chopVersionNumber(s1)
+	ss2 := chopVersionNumber(s2)
+	return ss1 == ss2
+}
+
+func chopVersionNumber(s string) string {
```

Comment:
```suggestion
// Strips a suffix "-digit*[.digit*]*" from the given string if present. Otherwise returns
// the string unmodified.
func chopVersionNumber(s string) string {
```